File: algorithms/validate_bst .py
def validateBst(tree, minValue = float("-inf"), maxValue = float("inf")):
    if tree == None: 
        return True
    if tree.value < minValue or tree.value >= maxValue: 
        return False
    return validateBst(tree.left, minValue, tree.value) and validateBst(tree.right, tree.value, maxValue)

File: algorithms/javascript/validate_bst.js
function validateBst(tree){
    return validateBstHelper(tree, -Infinity, Infinity);
}

function validateBstHelper(tree, minValue, maxValue) {
    if(tree == null) return true;
    if(tree.value < minValue || tree.value >= maxValue) return false;
    return validateBstHelper(tree.left, minValue, tree.value) && validateBstHe(tree.right, tree.value, maxValue);
}